x/gamm/genesis.go

```diff
@@ -44,11 +42,7 @@ func ExportGenesis(ctx sdk.Context, k keeper.Keeper) *types.GenesisState {
 	}
 	poolAnys := []*codectypes.Any{}
 	for _, poolI := range pools {
-		pool, ok := poolI.(*types.BalancerPool)
-		if !ok {
-			panic(fmt.Errorf("pool (%d) is not basic pool", poolI.GetId()))
-		}
-		any, err := codectypes.NewAnyWithValue(pool)
```

Comment:
Just want to double check, theres no reason we should have this restriction right?